diff --git a/RenderSystems/GL/src/GLX/OgreGLXWindow.cpp b/RenderSystems/GL/src/GLX/OgreGLXWindow.cpp
--- RenderSystems/GL/src/GLX/OgreGLXWindow.cpp
+++ RenderSystems/GL/src/GLX/OgreGLXWindow.cpp
@@ -401,16 +401,7 @@
 			
 			glxDrawable = mWindow;
 			
-			XMapWindow(xDisplay, mWindow);
-			
-			if (mIsFullScreen && !hidden)
-			{
-				switchFullScreen (true);
-			}
-			else if (hidden)
-			{
-				setHidden(true);
-			}
+			setHidden(hidden);
 			XFlush(xDisplay);
 			
 			WindowEventUtilities::_addRenderWindow(this);
